Use C++20 std::ranges::fill in WTF

iidmsa opened the pull request:
#### 57cd7a45bfda4ff6dd59d01c051ef880be2dd78e
<pre>
Use C++20 std::ranges::fill in WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=302057">https://bugs.webkit.org/show_bug.cgi?id=302057</a>
<a href="https://rdar.apple.com/164136608">rdar://164136608</a>

Reviewed by Chris Dumez.

* Source/WTF/wtf/TrailingArray.h:
(WTF::TrailingArray::fill):
* Source/WTF/wtf/Vector.h:
(WTF::Malloc&gt;::fill):
* Source/WTF/wtf/text/TextBreakIterator.h:
(WTF::CachedLineBreakIteratorFactory::PriorContext::reset):

Canonical link: <a href="https://commits.webkit.org/302762@main">https://commits.webkit.org/302762@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/59963a2ab3e85b3956528e2c213a1cf04b50d841

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129737 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1998 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40593 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137126 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81203 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1947 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1887 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98831 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66655 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132684 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1487 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116193 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79509 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1404 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34326 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80398 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/121729 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109883 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34828 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139608 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/128189 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1792 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1677 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107339 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1837 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112542 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107214 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27365 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1449 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31035 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54554 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1865 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/65234 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/161203 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1679 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40197 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1714 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1786 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Jarred-Sumner commented on Source/WTF/wtf/Vector.h:
```diff
@@ -1113,7 +1113,7 @@ void Vector<T, inlineCapacity, OverflowHandler, minCapacity, Malloc>::fill(const
     asanBufferSizeWillChangeTo(newSize);
 
-    std::fill(begin(), end(), val);
+    std::ranges::fill(*this, val);
```
This might have a negative performance impact - note the TODO comments: https://github.com/gcc-mirror/gcc/blob/cc3ade6da5c4b7d55425a15f25325ae84ef945a2/libstdc%2B%2B-v3/include/bits/ranges_algobase.h#L592-L631

[Also relevant](https://lemire.me/blog/2025/10/05/stdranges-may-not-deliver-the-performance-that-you-expect/)

cdumez replied:
I am doing some performance A/B testing and will report back. If it doesn't impact our benchmarks, I think we should land this as it makes the code simpler / more concise.

cdumez reviewed (approved):
No performance impact. LGTM

iidmsa committed:
Use C++20 std::ranges::fill in WTF
https://bugs.webkit.org/show_bug.cgi?id=302057
rdar://164136608

Reviewed by Chris Dumez.

* Source/WTF/wtf/TrailingArray.h:
(WTF::TrailingArray::fill):
* Source/WTF/wtf/Vector.h:
(WTF::Malloc>::fill):
* Source/WTF/wtf/text/TextBreakIterator.h:
(WTF::CachedLineBreakIteratorFactory::PriorContext::reset):

Canonical link: https://commits.webkit.org/302762@main
